plugins/autolink/plugin.js

```diff
@@ -64,6 +115,55 @@
 				}
 				return data;
 			}
+
+			function getHtmlToInsert( text ) {
+				var link = text.replace( doubleQuoteRegex, '%22' ),
+					template = link.match( CKEDITOR.config.autolink_urlRegex ) ?
+					urlTemplate.output( { text: link } )
+					: emailTemplate.output( { text: link } );
+
+				return tryToEncodeLink( template );
+			}
+
+			function matchCallback( text, offset ) {
+				var query = text.slice( 0, offset ),
+					// Remove empty strings.
+					parts = query.split( /(\s+)/g ),
+					linkText = parts[ parts.length - 1 ];
+
+				if ( !linkText ) {
+					return null;
+				}
+
+				var match = linkText.match( CKEDITOR.config.autolink_urlRegex ) ||
+					linkText.match( CKEDITOR.config.autolink_emailRegex );
+
+				if ( !match ) {
+					return null;
+				}
+
+				return { start: query.indexOf( linkText ), end: offset };
+			}
 		}
 	} );
+
+	/**
+	 * Regex used by [Autolink](https://ckeditor.com/cke4/addon/autolink) plugin to match URL adresses.
+	 *
+	 * @cfg {RegExp} [autolink_urlRegex]
+	 * @since 4.11.0
+	 * @member CKEDITOR.config
+	 */
+	CKEDITOR.config.autolink_urlRegex = /^(https?|ftp):\/\/(-\.)?([^\s\/?\.#]+\.?)+(\/[^\s]*)?[^\s\.,]$/ig;
+	// Regex by Imme Emosol.
+
+	/**
+	 * Regex used by [Autolink](https://ckeditor.com/cke4/addon/autolink) plugin to match email adresses.
+	 *
+	 * @cfg {RegExp} [autolink_emailRegex]
+	 * @since 4.11.0
+	 * @member CKEDITOR.config
+	 */
+	CKEDITOR.config.autolink_emailRegex = /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/g;
+	// Regex by (https://www.w3.org/TR/html5/forms.html#valid-e-mail-address).
```

Comment:
Actually it's no longer in HTML 5.x, now it's only in [HTML LS](https://html.spec.whatwg.org/#e-mail-state-(type=email)).